### PR TITLE
Changes pegasus to respect deleted sections and followers.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Note that deleted_at was introduced in #7871 and #7917. An upcoming PR will add `acts_as_paranoid` to the sections and followers dashboard models.

Also changes pegasus to (further) respect deleted users, where missed in #7899.

Note that pegasus/helpers/professional_development_workshop_helpers.rb is explictly not being updated as it is in the process of being deprecated (per @aoby). Further note that /v2/sections/ API (also pegasus APIs more generally) remains woefully under-tested. Though this PR helps improve test coverage, significantly more needs to be done.

### CODE REVIEW

File: pegasus/test/test_v2_user_routes.rb [hunk 1]
```diff
@@ -166,18 +165,22 @@ def expected_v2_students_id_hash_for(user)
 
       it 'returns 403 "Forbidden" when signed in as another student' do
         with_role FakeDashboard::SELF_STUDENT
-        @pegasus.post "/v2/students/#{FakeDashboard::STUDENT[:id]}/update",
-          {name: NEW_NAME}.to_json,
-          'CONTENT_TYPE' => 'application/json;charset=utf-8'
-        assert_equal 403, @pegasus.last_response.status
+        Dashboard.db.transaction(rollback: :always) do
```
Comment (aoby) [inline pegasus/test/test_v2_user_routes.rb:174]: You can get this for free by deriving from [SequelTestCase](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/test/sequel_test_case.rb)

**Comment by aoby** [comment]
lgtm, aside from above suggestion

**Comment by ashercodeorg** [comment]
Assuming tests pass, now ready for review.

CC @islemaster @Hamms

File: pegasus/test/test_v2_section_routes.rb [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+require 'mocha/mini_test'
+require_relative 'fixtures/fake_dashboard'
+require_relative 'fixtures/mock_pegasus'
+
+class V2SectionRoutesTest < Minitest::Test
+  describe 'Section Routes' do
+    before do
+      FakeDashboard.use_fake_database
+      $log.level = Logger::ERROR # Pegasus spams debug logging otherwise
+      @pegasus = Rack::Test::Session.new(Rack::MockSession.new(MockPegasus.new, "studio.code.org"))
```
Comment (islemaster) [inline pegasus/test/test_v2_section_routes.rb:14]: Consider using the `build_rack_mock_session` approach Dave pointed out to me [in this PR](https://github.com/code-dot-org/code-dot-org/pull/7911#discussion_r60297776).  It's not well-documented, but basically it provides all the session request methods (`get`, `post`, `last_response`, etc) in the scope of the test.

File: pegasus/test/test_v2_section_routes.rb [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+           "course"=>nil,
+           "teachers"=>[{"id"=>7, "location"=>"/v2/users/7"}],
+           "students"=>[{"student_user_id"=>5, "location"=>"/v2/users/", "age"=>nil, "completed_levels_count"=>0}]
+          }],
+          JSON.parse(@pegasus.last_response.body)
+      end
+    end
+
+    describe 'POST /v2/sections' do
+      # TODO(asher).
```
Comment (islemaster) [inline pegasus/test/test_v2_section_routes.rb:82]: TODO?

File: pegasus/test/test_v2_section_routes.rb [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            "name"=>"Fake Section A",
+            "login_type"=>"email",
+            "grade"=>nil,
+            "code"=>nil
+          }],
+          JSON.parse(@pegasus.last_response.body)
+      end
+
+      it 'ignores deleted sections' do
+        # TODO(asher).
```
Comment (islemaster) [inline pegasus/test/test_v2_section_routes.rb:109]: TODO?

**Comment by islemaster** [comment]
Nice testing :+1:.  I see several empty tests with TODOs, just checking whether you intend to address those now or in a near-future PR.

**Comment by ashercodeorg** [comment]
In a near-future PR. Otherwise, I could spend days writing tests.

File: pegasus/test/test_v2_section_routes.rb (hunk 2 above)
Comment (ashercodeorg) [inline pegasus/test/test_v2_section_routes.rb:14]: If it is all the same to you, I'll update the various tests with this paradigm in a separate PR.

File: pegasus/test/test_v2_user_routes.rb (hunk 1 above)
Comment (ashercodeorg) [inline pegasus/test/test_v2_user_routes.rb:174]: Likewise, if it is all the same, I'll update all the tests to derive from SequelTestCase in a followup PR.

**Pull request merged**